lib/src/main/java/dev/openfeature/javasdk/HookSupport.java

```diff
@@ -0,0 +1,80 @@
+package dev.openfeature.javasdk;
+
+import java.util.*;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+
+import com.google.common.collect.Lists;
+import lombok.RequiredArgsConstructor;
+import org.slf4j.Logger;
+
+@SuppressWarnings({"unchecked", "rawtypes", "PMD.LoggerIsNotStaticFinal"})
+@RequiredArgsConstructor
+class HookSupport {
+
+    private final Logger log;
+
+    public void errorHooks(FlagValueType flagValueType, HookContext hookCtx, Exception e, List<Hook> hooks, Map<String, Object> hints) {
+        executeHooks(flagValueType, hooks, "error", hook -> hook.error(hookCtx, e, hints));
+    }
+
+    public void afterAllHooks(FlagValueType flagValueType, HookContext hookCtx, List<Hook> hooks, Map<String, Object> hints) {
+        executeHooks(flagValueType, hooks, "finally", hook -> hook.finallyAfter(hookCtx, hints));
+    }
+
+    public void afterHooks(FlagValueType flagValueType, HookContext hookContext, FlagEvaluationDetails details, List<Hook> hooks, Map<String, Object> hints) {
+        executeHooksUnchecked(flagValueType, hooks, hook -> hook.after(hookContext, details, hints));
+    }
+
+    private <T> void executeHooks(
+        FlagValueType flagValueType, List<Hook> hooks,
+        String hookMethod,
+        Consumer<Hook<T>> hookCode
+    ) {
+        hooks
+            .stream()
+            .filter(hook -> hook.supportsFlagValueType() == flagValueType)
```

Comment:
Should `supportsFlagValueType` take a type as an argument? Seems plausible that a hook may want to support multiple.

Reply:
Well, then it doesn't work with typed Hooks anymore. There can either be a `Hook<String>` or a `Hook<Boolean>` but not a combined one, unless a generic `Hook<?>` is used. 
This is actually a miss on my side, as the generic hooks are currently not called on boolean, string and integer values.

Reply:
What I'm suggesting is this line looks like:

```
             .filter(hook -> hook.supportsFlagValueType(flagValueType)) 
```

The `Hook<String>` implementation would look like:

```
public boolean supportsFlagValueType(FVT type) {
  return type == TYPES.String;
}
```

but this opens the possibility for generic flags to specify the types they want to work with which seems useful?

Reply:
I changed it but I'm currently not sure, whether I like it. A pro is certainly, that the Hook now decides, whether it can handle a feature flag type.